apps/site/pages/docs/api-extensions/troubleshooting.mdx

```diff
@@ -0,0 +1,77 @@
+---
+title: 'Troubleshooting'
+sidebar_label: 'API Extensions'
+---
+
+import { Callout, Step, Steps } from 'nextra-theme-docs'
+import { ExtendableQueryTable } from './../../../components/ExtendableQueryTable'
+
+<header>
+
+# Troubleshooting
+
+</header>
+
+## GraphQL changes not visible during development
+If your GraphQL changes aren't appearing during development, this is because the changes you have made since you started the development server (`yarn dev`) are probably not optimized yet. 
+Follow these steps to trigger the optimization:
+
+- Run `yarn generate` (recommended) or `yarn run faststore generate-graphql`.
+
+    <Callout type="info" emoji="ℹ️">
+        This optimization can also be performed while the development server is running.
+    </Callout>
+
+
+- Alternatively, you can stop the development server and then restart it using `yarn dev`.
+
+---
+
+## Deploy preview/production GraphQL schema different from development
+
+If you notice differances between the GraphQL schema in your deploy preview or production environment compared to your development setup, it may be due to the schema not being optimized since the development server's initiation. 
+Typically, the build process optimizes the schema before deployment to ensure it accurately reflects the schema declared in the store's code.
+
+Refer to the [GraphQL changes not visible during development](/docs/api-extension/troubleshooting#graphql-changes-not-visible-during-development) topic, to fix the issue.
+
+---
+
+## Generation errors and warnings
```

Comment:
I'd move this section up, after "GraphQL changes not visible during development". It will be way more common than "Deploy preview/production GraphQL schema different from development".